clusters/build-clusters/01_cluster/hack/upgrade-cluster.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if [[ "$#" -ne 1 ]]; then
+  echo "You must enter the version to upgrade, e.g., quay.io/openshift-release-dev/ocp-release:4.4.0-rc.12-x86_64"
+  exit 1
+fi
+
+CHOSEN_VERSION=$1
+readonly CHOSEN_VERSION
+
+CHOSEN_IMAGE="$(echo ${CHOSEN_VERSION} | cut -d':' -f1)"
+readonly CHOSEN_IMAGE
+
+CHOSEN_TAG="$(echo ${CHOSEN_VERSION} | cut -d':' -f2)"
+readonly CHOSEN_TAG
+
+cmd="echo"
+if [[ "${DRY_RUN:-}" == "false" ]]; then
+  cmd="oc"
+fi
+
+if [[ "${CHOSEN_TAG}" == *"nightly"* ]]; then
```

Comment:
There are two flavors of nightlies:

* Things like `registry.svc.ci.openshift.org/ocp/release:4.4.0-0.nightly-2020-04-27-013217` from from [the release-controller][1], that you need to pin to avoid garbage collection.
* Things like `quay.io/openshift-release-dev/ocp-release-nightly@sha256:3453f60569f5354aa19381562ed513647bd22e0d405cb9b94c35073366c6818e` from [the mirrors][2] that are never garbage collected.

It's generally better to use the latter when you need a nightly.  Besides the lack of GC, you can also run them with a vanilla pull secret, while the release-controller pullspecs require an additional CI-registry token (which then needs to be refreshed ~monthly).  I'd recommend dropping this block, and then sticking to mirror nightlies in the future, if you need to drop into nightlies at all.

[1]: https://openshift-release.svc.ci.openshift.org/releasestream/4.4.0-0.nightly/release/4.4.0-0.nightly-2020-04-27-013217
[2]: https://mirror.openshift.com/pub/openshift-v4/clients/ocp-dev-preview/4.4.0-0.nightly-2020-04-27-013217/release.txt

Reply:
Updated.
I have never used the former though.

Reply:
Removed it for now.
I need to double check why the token never expires (probably i used a token from a SA).
Also need to check the cluster has the token from `quay.io/openshift-release-dev/ocp-release-nightly`.

Reply:
Checked ...

```
oc --context build01 get secret -n openshift-config pull-secret -o yaml | yq -r '.data.".dockerconfigjson"' | base64 -d | jq -r '.auths | keys[]'
cloud.openshift.com
quay.io
quay.io/openshift-cnv-dev
registry.connect.redhat.com
registry.redhat.io
registry.svc.ci.openshift.org
```

And I copied the secret at the installation time:

https://github.com/openshift/release/blob/63a47319f50a34ab249235c34d38c9b89641972c/clusters/build-clusters/01_cluster/Makefile#L9